### PR TITLE
Return the unsubscribe from the subscriber

### PR DESCRIPTION
**Pull request by cristianbote**
This introduces the return of unsubscriber, when doing a subscribe.

```js
let store = createStore();
let unsub = store.subscribe(state => modifyState(state));

// After you're done just call the unsubscriber from 'unsub'
unsub();
```

I often found that this little help goes a long way. Sure it comes with increase in size. Bumps it to `680B` but maybe you think it's worth it?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -117,7 +117,7 @@ Creates a new store, which is a tiny evented state container.
 let store = createStore();
    store.subscribe( state => console.log(state) );
    store.setState({ a: 'b' });   // logs { a: 'b' }
-   store.setState({ c: 'd' });   // logs { a: 'b', c: 'd' }
+   store.setState({ c: 'd' });   // logs { c: 'd' }
```
Comment (cristianbote) [inline README.md]: This is being addressed here: #9

File: README.md [hunk 2]
```diff
@@ -153,7 +155,7 @@ Remove a previously-registered listener function.
 
 ##### getState
 
-Retrieve the current state object.
+Retreive the current state object.
```
Comment (cristianbote) [inline README.md]: Same, fixed in #9

**Comment by kadmil** [comment]
Love this api, as you could skip storing your subscriber lambda in separate variable

**Comment by developit** [comment]
The currently implementation relies on `this`, which I'd prefer to avoid.  Can we either declare `store = {}` in the `createStore()` scope, hoist `unsubscribe()` as an inner function there, or inline `unsubscribe()` into the returned anonymous function?

**Comment by cristianbote** [comment]
Yup. Initially that was my thinking, just inline the `unsubscribe` and maybe remove it from the api. But, didn't want to introduce breaking changes on day two 😀.

I'll update it, soon.

**Pull request merged**